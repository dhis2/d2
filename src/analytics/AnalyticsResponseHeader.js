class AnalyticsResponseHeader {
    constructor(header = {}, options = { isPrefix: false, isCollect: false, index: undefined }) {
        Object.assign(this, header, options);
    }

    getIndex() {
        return this.index;
    }

    setIndex(value) {
        const index = +value;

        if (!Number.isNaN(index) && Number.isFinite(index)) {
            this.index = parseInt(index, 10);
        }
    }
}

export default AnalyticsResponseHeader;
