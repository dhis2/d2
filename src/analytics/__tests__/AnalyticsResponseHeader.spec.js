import AnalyticsResponseHeader from '../AnalyticsResponseHeader'

let responseHeader
let expectedResponseHeader

describe('AnalyticsResponseHeader', () => {
    beforeEach(() => {
        responseHeader = new AnalyticsResponseHeader()
        expectedResponseHeader = {}
    })

    describe('constructor', () => {
        const flags = {
            isPrefix: false,
            isCollect: false,
            index: undefined,
        }

        it('should not be allowed to be called without new', () => {
            expect(() =>
                AnalyticsResponseHeader()
            ).toThrowErrorMatchingSnapshot()
        })

        it('should initialize properties', () => {
            expect(responseHeader.isPrefix).toEqual(flags.isPrefix)
            expect(responseHeader.isCollect).toEqual(flags.isCollect)
            expect(responseHeader.index).toEqual(flags.index)
        })

        it('should set the header when passed as argument', () => {
            const header = {
                name: 'cejWyOfXge6',
                column: 'Gender',
                valueType: 'TEXT',
                type: 'java.lang.String',
                hidden: false,
                meta: true,
                optionSet: 'pC3N9N77UmT',
            }

            responseHeader = new AnalyticsResponseHeader(header)

            expectedResponseHeader = { ...header, ...flags }

            Object.keys(expectedResponseHeader).forEach(key => {
                expect(responseHeader[key]).toEqual(expectedResponseHeader[key])
            })
        })
    })

    describe('properties', () => {
        describe('.setIndex()', () => {
            it('should set the index with the given value', () => {
                responseHeader.setIndex(12)

                expect(responseHeader.index).toEqual(12)
            })

            it('should set the index as numeric when passing a string', () => {
                responseHeader.setIndex('10')

                expect(responseHeader.index).toEqual(10)
            })

            it('should set the index as integer', () => {
                responseHeader.setIndex(12.5)

                expect(responseHeader.index).toEqual(12)
            })
        })

        describe('.getIndex()', () => {
            it('should return the correct index value', () => {
                responseHeader.index = 17

                expect(responseHeader.getIndex()).toEqual(17)
            })
        })
    })
})
