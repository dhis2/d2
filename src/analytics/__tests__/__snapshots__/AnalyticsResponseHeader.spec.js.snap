// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsResponseHeader constructor should not be allowed to be called without new 1`] = `"Class constructor AnalyticsResponseHeader cannot be invoked without 'new'"`;
